client/packages/common/src/ui/layout/tables/components/Cells/StatusCell/StatusCell.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+import { CellProps } from '../../../columns';
+import { Box, CircleIcon, Typography } from '@openmsupply-client/common';
+import { RecordWithId } from '@common/types';
+
+interface StatusProps {
+  statusMap: Record<
+    string,
+    { label: string; color: string; bgColor?: string }
+  > | null;
+}
+
+export const StatusCell = <T extends RecordWithId>({
+  rowData,
+  column,
+  statusMap,
+}: CellProps<T> & StatusProps) => {
+  const { label, color, bgColor } =
+    statusMap?.[String(column.accessor({ rowData }))] ?? {};
+
+  if (!label) return null;
+  return (
+    <Box
+      sx={{
+        borderRadius: 4,
+        textAlign: 'center',
+        backgroundColor: bgColor,
+      }}
+      paddingY={0.1}
+      paddingX={0.5}
+      display="flex"
+      alignItems="center"
+      width="fit-content"
+      minWidth={100}
```

Comment:
Nit, but I'm not sure the min width is needed, I think it would look nicer just fitting the contents?

<img width="170" alt="Screenshot 2024-09-12 at 3 49 20 PM" src="https://github.com/user-attachments/assets/e7db5ed6-001c-417a-b742-67ea243ec047">